src/50_Samples_%26_Templates/Product.html

```diff
@@ -387,7 +387,9 @@ <h2 class="m1">Apple</h2>
    <!-- ## Image Gallery -->
   <!-- The `amp-carousel` component works very well for product image galleries. Learn how the `amp-carousel` component works [here](/advanced/image_galleries_with_amp-carousel/).
 
-  Notice how we are binding to the `slide` and `class` by using to the values of the variables `selectedSlide` and `selectedColor`. -->
+  Notice how we are binding to the `slide` and `class` by using to the values of the variables `selectedSlide` and `selectedColor`. 
+  It's not possible to set a default value for variables, for example we use the expression`(!selectedColor || selectedColor == 'green')`
```

Comment:
Ok,
I tried to rewrite it this way:
`          <amp-selector name="color"
                  layout="container"
                  [selected]="product.selectedColor"
                  on="select:AMP.setState({
                    {product: {
                      selectedColor: event.targetOption,
                      selectedSlide: 0,
                      selectedSize: product[event.targetOption].sizes[selectedSize] != 'unavailable' ? selectedSize : product[event.targetOption].defaultSize
                    }
                  })">`
but still I get:

> error.js:128 Invalid action definition in  
<amp-selector name=​"color" layout=​"container" [selected]=​"product.selectedColor" on=​"select:​AMP.setState({
                    {product:​ {
                      selectedColor:​ event.targetOption,
                      selectedSlide:​ 0,
                      selectedSize:​ product[event.targetOption]​.sizes[selectedSize]​ != 'unavailable' ? selectedSize :​ product[event.targetOption]​.defaultSize
                    }​
                  }​)​" class=​"i-amphtml-element i-amphtml-layout-container i-amphtml-layout i-amphtml-error" role=​"listbox" selected=​"green">​…​</amp-selector>​
 : [ select:AMP.setState({
                    {product: {
                      selectedColor: event.targetOption,
                      selectedSlide: 0,
                      selectedSize: product[event.targetOption].sizes[selectedSize] != 'unavailable' ? selectedSize : product[event.targetOption].defaultSize
                    }
                  }) ] ; unexpected token []